federation/schema.go

```diff
@@ -196,14 +192,45 @@ func ConvertVersionedSchemas(schemas serviceSchemas) (*SchemaWithFederationInfo,
 
 		serviceSchema, err := mergeSchemaSlice(versionSchemas, Intersection)
 		if err != nil {
-			return nil, err
+			return nil, nil, nil, err
 		}
 
 		serviceSchemasByName[service] = serviceSchema
-
 		serviceSchemas = append(serviceSchemas, serviceSchema)
 	}
 
+	return serviceNames, serviceSchemas, serviceSchemasByName, nil
+}
+
+// MergeSchemas takes the introspection query results for all
+// verions of all executor services, and merges them into a
+// single introspection query result that represents the full
+// federated schema.
+func MergeSchemas(schemas serviceSchemas) (*IntrospectionQueryResult, error) {
```

Comment:
Maybe rename this function to `MergeIntrospectionSchema`? `MergeSchema` feels merge the full schemas to me.